### DISCUSSION TITLE
Recommended behavior when a marker lacks a required argument

### DISCUSSION THREAD

**Discussion by jwodder** [pytest-dev/pytest#9890] [Q&A]
I'm currently writing a pytest plugin that defines a custom marker that takes an argument and doesn't make any sense to be used without an argument.  What do you recommend my plugin to do if a user omits the argument from the marker?  Should it do nothing (as though the marker weren't there), or should it raise an error from inside the pytest hook that processes the marker (and, if so, what would be the best way to do that)?

**Comment by RonnyPfannschmidt** [comment]
Depending on where the marker is used, either value/type errors or a Usage Error

**Reply by jwodder** [reply]
The marker is just applied to test functions.  Assuming I raise a `TypeError`, how would I test this?  `pytester` seems to convert any exceptions inside `pytest_runtest_makereport` to an INTERNALERROR.
